core/base/test/TStringTest.cxx

```diff
@@ -0,0 +1,18 @@
+#include "gtest/gtest.h"
+
+#include "ROOTUnitTestSupport.h"
+
+#include "TString.h"
+
+TEST(TString, Basics)
+{
+   TString *s = nullptr;
+   ROOT_EXPECT_ERROR(s = new TString("Test", -5), "TString::TString", "Negative length!");
+   EXPECT_STREQ("", s);
```

Comment:
ok probably the last thing, this line should be removed and a `delete s` should be added (it destroys the TString object we created with `new`, which is leaked otherwise. pointers...):

```suggestion
   delete s;
```

Reply:
Okay. I committed the change